packages/asset-service/src/generateAssetData/avalanche/avalanche.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  avalancheAssetId,
+  avalancheChainId,
+  CHAIN_NAMESPACE,
+  CHAIN_REFERENCE,
+  toAssetId
+} from '@shapeshiftoss/caip'
+import { Asset } from '@shapeshiftoss/types'
+import axios from 'axios'
+
+type AvalancheToken = {
+  nativeNetwork: string
+  nativeContractAddress: string
+  denomination: number // precision
+  chainlinkFeedAddress: string
+  logo: string
+  coingeckoId: string
+}
+
+type AvalancheTokenList = Record<string, AvalancheToken>
+
+type AvalancheContractAddress = {
+  [symbolDotE: string]: string // value is avalanche contract address
+}
+
+type GetAvalancheAssets = () => Promise<Asset[]>
+
+export const getAvalancheAssets: GetAvalancheAssets = async () => {
+  // this file contains the eth side of tokens bridged to avalanche
+  // see https://docs.penguinfinance.io/getting-started/new-.e-tokens-explainer for more info
+  // https://github.com/ava-labs/avalanche-bridge-resources/blob/main/token_list.json
+  const avalancheTokenListUrl =
+    'https://rawcdn.githack.com/ava-labs/avalanche-bridge-resources/31fabeeeca69fa774be54c5b6ec93bea56161d6a/token_list.json'
+  const { data: tokenList } = await axios.get<AvalancheTokenList>(avalancheTokenListUrl)
+
+  // https://github.com/ava-labs/avalanche-bridge-resources/blob/main/avalanche_contract_address.json
+  const avalancheContractAddressesUrl =
+    'https://rawcdn.githack.com/ava-labs/avalanche-bridge-resources/31fabeeeca69fa774be54c5b6ec93bea56161d6a/avalanche_contract_address.json'
+  const { data: contractAddresses } = await axios.get<AvalancheContractAddress>(
+    avalancheContractAddressesUrl
+  )
+
+  const chainNamespace = CHAIN_NAMESPACE.Ethereum
+  const chainReference = CHAIN_REFERENCE.AvalancheCChain
+
+  const chainId = avalancheChainId
+  const explorer = 'https://snowtrace.io/'
+  const explorerAddressLink = `${explorer}address`
+  const explorerTxLink = `${explorer}tx`
+
+  const assets = Object.entries(tokenList).reduce<Asset[]>((acc, [symbol, v]) => {
+    const name = `${symbol} on Avalanche`
+    const precision = v.denomination
+    const icon = v.logo
+    const color = '#FFFFFF' // this will get picked up by the color generation script
+    const assetReference = contractAddresses[`${symbol}.e`].toLowerCase()
+    const assetNamespace = 'erc20'
```

Comment:
i find it silly to add the extra vars just for the js key=value syntax when not used elsewhere. more readable imo to have the value being defined in the object so you don't have to look for where it was defined somewhere else